src/extensibility/ExtensionManager.js

```diff
@@ -99,6 +99,25 @@ define(function (require, exports, module) {
      *         status: the current status, one of the status constants above
      */
     var extensions = {};
+    
+    /**
+     * @private
+     * @type {Object.<string, {metadata: Object, path: string, status: string}>}
+     * The set of all extensions (registry and locally installed) present in user location.
+     * The fields of each record are:
+     *     registryInfo: object containing the info for this id from the main registry (containing metadata, owner,
+     *         and versions).
+     *     installInfo: object containing the info for a locally-installed extension:
+     *         metadata: the package metadata loaded from the local package.json, or null if it's a legacy extension.
+     *         path: the local path to the extension folder on disk
+     *         status: the current status, one of the status constants above
+     */
+    var userExtensions = {};
+    
+    /**
+     * Registry object container  
+     */
+    var registryObject = {};
```

Comment:
@prafulVaishnav I'm still not convinced this change was needed.  The existing `extensions` object includes both sets of data with clean separation.  You're just duplicating all that data in two other objects, created with code that is largely duplicated code too.

Is there something that wouldn't work right if `HealthDataUtils.getInstalledExtensions()` just did this?

```
        _.forEach(ExtensionManager.extensions, function (extension, extensionId) {
            if (extension.installInfo && extension.registryInfo) {
                userInstalledExtensions.push({"name" : extensionId, "version" : extension.installInfo.metadata.version});
            }
        });
```